.github/workflows/release_prepare_pr.yaml

```
@@ -0,0 +1,99 @@
+name: Prepare Release PR
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+concurrency:
+  # Do not cancel jobs on main by forcing a unique group name
+  group: ${{ github.workflow }}-${{ github.ref_name == 'main' && github.run_id || github.ref_name }}
```

```suggestion
  group: ${{ github.workflow }}
```

In this case, there are no two different contexts (PRs vs push to main), but only one.
I think we're OK with cancelling jobs - only the `HEAD` of the `main` is important anyway.